### PR TITLE
add babelflow-based parallel merge tree (pmt) package

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return args
+
+    def cmake_install(self, spec, prefix):
+        if "+shared" in spec:
+            cmake_args.append('-DBUILD_SHARED_LIBS=ON')
+        else:
+            cmake_args.append('-DBUILD_SHARED_LIBS=OFF')
+
+        make()
+        make('install')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/pmt/package.py]: bug fix:
```suggestion
    def cmake_args(self):
        args = [
            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
                'ON' if '+shared' in spec else 'OFF'),
            '-DLIBRARY_ONLY=ON'
        ]
                
        return args
```

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class Pmt(CMakePackage):
+    """A multi-runtime implementation of a distributed merge tree
+    segmentation algorithm. The implementation relies on the framework
+    BabelFlow, which allows to execute the algorithm on different runtime
+    systems."""
+
+    homepage = "https://bitbucket.org/cedmav/parallelmergetree"
+    url      = "https://bitbucket.org/cedmav/parallelmergetree/get/ascent.zip"
+
+    maintainers = ['spetruzza']
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/pmt/package.py]: cc @spetruzza for review :)

**Comment by cyrush** [comment]
thanks @ax3l  I applied the suggest patch

**Review by ax3l — APPROVED** [review]

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class Pmt(CMakePackage):
+    """A multi-runtime implementation of a distributed merge tree
+    segmentation algorithm. The implementation relies on the framework
+    BabelFlow, which allows to execute the algorithm on different runtime
+    systems."""
+
+    homepage = "https://bitbucket.org/cedmav/parallelmergetree"
+    url      = "https://bitbucket.org/cedmav/parallelmergetree/get/ascent.zip"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmt/package.py]:
```suggestion
    git      = "https://bitbucket.org/cedmav/parallelmergetree.git"
```
Then you won't need to specify it below.

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    maintainers = ['spetruzza']
+
+    version('develop',
+            git='https://bitbucket.org/cedmav/parallelmergetree.git',
+            branch='ascent',
+            commit='6774ed74fd13b9747ac792978a676ce6e8b05cab',
+            submodules=True,
+            preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmt/package.py]: No need for preferred if there's only a single version.

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    homepage = "https://bitbucket.org/cedmav/parallelmergetree"
+    url      = "https://bitbucket.org/cedmav/parallelmergetree/get/ascent.zip"
+
+    maintainers = ['spetruzza']
+
+    version('develop',
+            git='https://bitbucket.org/cedmav/parallelmergetree.git',
+            branch='ascent',
+            commit='6774ed74fd13b9747ac792978a676ce6e8b05cab',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmt/package.py]: Do branches and commits work together? I've only ever seen one or the other in use.

Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: I checked this out, it turns out they are both the same -- so at least there is no mystery about which is right. 
I am going to stick with the commit.

File: var/spack/repos/builtin/packages/pmt/package.py (hunk 4 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: thanks  -- fixed this

File: var/spack/repos/builtin/packages/pmt/package.py (hunk 3 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: thanks  -- fixed this

File: var/spack/repos/builtin/packages/pmt/package.py (hunk 5 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: fixed this

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/pmt/package.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Pmt(CMakePackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmt/package.py]: Any reason not to call this parallelmergetree?

Comment (spetruzza) [reply var/spack/repos/builtin/packages/pmt/package.py]: Not really, just to make it short. For consistency with the repository name we could rename it to parallelmergetree, @cyrush what do you think (this change has to be reflected also in the Ascent uberenv variant [dependencies](https://github.com/Alpine-DAV/ascent/blob/develop/scripts/uberenv/packages/ascent/package.py#L120))?

Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: @spetruzza -- I am fine with this change, I'll make it when I rebase now that babelflow is in. The uberenv path will be different for a bit until we can cut over to using new spack, so we will have to be mindful of that.

Comment (cyrush) [reply var/spack/repos/builtin/packages/pmt/package.py]: I made this change.

File: var/spack/repos/builtin/packages/parallelmergetree/package.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class ParallelMergeTree(CMakePackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/parallelmergetree/package.py]:
```suggestion
class Parallelmergetree(CMakePackage):
```

**Comment by adamjstewart** [comment]
@alalazo any idea why the unit tests are failing? The last commit (`ParallelMergeTree` -> `Parallelmergetree`) should have fixed the issue being reported by the unit tests. Is this another case where we are checking out the wrong commit?

**Comment by scheibelp** [comment]
Close/reopen to restart tests

**Pull request merged**